tests/proxy/lock_test.py

```diff
@@ -80,4 +80,41 @@ def test_force_unlock(self):
         self.assertFalse(self.lock.is_locked())
 
     def test_str(self):
-        self.assertTrue(str(self.lock).startswith("Lock"))
\ No newline at end of file
+        self.assertTrue(str(self.lock).startswith("Lock"))
+
+    def test_key_owner_shutdowns_after_invocation_timeout(self):
+        key_owner = self.cluster.start_member()
+
+        invocation_timeout_seconds = 1
+        self.client.invoker.invocation_timeout = invocation_timeout_seconds
+
+        key = generate_key_owned_by_instance(self.client, key_owner.address)
+        server_lock = self.client.get_lock(key).blocking()
+        server_lock.lock()
+
+        e = Event()
+
+        def lock_thread_func():
+            lock = self.client.get_lock(key).blocking()
+            lock.lock()
+            lock.unlock()
+            e.set()
+
+        self.start_new_thread(lock_thread_func)
+
+        time.sleep(2 * invocation_timeout_seconds)
+        key_owner.shutdown()
+
+        partition_id = self.client.partition_service.get_partition_id(key)
+        while not (self.client.partition_service.get_partition_owner(partition_id) == self.member.address):
+            time.sleep(0.1)
+
+        self.assertTrue(server_lock.is_locked())
```

Comment:
If this assertion fails, will this code continue or abort ? If it aborts since the revert part wont be done other tests could fail as well.

Reply:
good point. fixed it